src/stream/src/common/table/state_table.rs

```diff
@@ -630,18 +644,24 @@ impl<S: StateStore> StateTable<S> {
         self.epoch = Some(new_epoch);
     }
 
-    pub async fn commit(&mut self, new_epoch: EpochPair) -> StreamExecutorResult<()> {
+    pub async fn commit(
+        &mut self,
+        new_epoch: EpochPair,
+        watermark: Option<&ScalarImpl>,
```

Comment:
can we use another function to update watermark to avoid hacking `commit`?

Reply:
> can we use another function to update watermark to avoid hacking `commit`?

after discussion with @Little-Wallace , we find that we need to use watermark to filter data in `commit`

Reply:
> Yes, but it does not matter to change the state table's interface, the state table can maintain the current watermark in memory by itself and filter data in `commit`

fixed